src/test/java/org/dcsa/ebl/controller/ShippingInstructionControllerTest.java

```diff
@@ -0,0 +1,212 @@
+    shipmentEquipmentTO.setActiveReeferSettings(activeReeferSettingsTO);
+    shipmentEquipmentTO.setIsShipperOwned(true);
+
+    PartyContactDetailsTO partyContactDetailsTO = new PartyContactDetailsTO();
+
+    PartyTO partyTO = new PartyTO();
+    partyTO.setPartyContactDetails(List.of(partyContactDetailsTO));
+
+    DocumentPartyTO documentPartyTO = new DocumentPartyTO();
+    documentPartyTO.setPartyFunction(PartyFunction.DDR);
+    documentPartyTO.setIsToBeNotified(true);
+    documentPartyTO.setParty(partyTO);
+
+    ReferenceTO referenceTO = new ReferenceTO();
+    referenceTO.setReferenceType(ReferenceTypeCode.EQ);
+    referenceTO.setReferenceValue("Some reference value");
+
+    shippingInstructionTO = new ShippingInstructionTO();
+    shippingInstructionTO.setCarrierBookingReference("XYZ12345");
+    shippingInstructionTO.setPlaceOfIssue(locationTO);
+    shippingInstructionTO.setShipmentEquipments(List.of(shipmentEquipmentTO));
+    shippingInstructionTO.setDocumentParties(List.of(documentPartyTO));
+    shippingInstructionTO.setReferences(List.of(referenceTO));
+    shippingInstructionTO.setIsShippedOnboardType(true);
+    shippingInstructionTO.setIsElectronic(true);
+    shippingInstructionTO.setIsToOrder(true);
+    shippingInstructionTO.setShippingInstructionID(UUID.randomUUID().toString());
+    shippingInstructionTO.setPlaceOfIssueID(locationTO.getId());
+    shippingInstructionTO.setAreChargesDisplayedOnCopies(true);
+    shippingInstructionTO.setAreChargesDisplayedOnOriginals(true);
+
+    OffsetDateTime now = OffsetDateTime.now();
+    shippingInstructionTO.setShippingInstructionCreatedDateTime(now);
+    shippingInstructionTO.setShippingInstructionUpdatedDateTime(now);
+
+    shippingInstructionResponseTO =
+        shippingInstructionMapper.dtoToShippingInstructionResponseTO(shippingInstructionTO);
+  }
+
+  @Test
+  @DisplayName(
+      "POST shipping-instructions should return 201 and valid shipping instruction json schema.")
+  void postShippingInstructionsShouldReturn201ForValidShippingInstructionRequest() {
+
+    ArgumentCaptor<ShippingInstructionTO> argument =
+        ArgumentCaptor.forClass(ShippingInstructionTO.class);
+
+    // mock service method call
+    when(shippingInstructionService.createShippingInstruction(any()))
+        .thenReturn(Mono.just(shippingInstructionResponseTO));
+
+    WebTestClient.ResponseSpec exchange =
+        webTestClient
+            .post()
+            .uri(SHIPPING_INSTRUCTION_ENDPOINT)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(BodyInserters.fromValue(shippingInstructionTO))
+            .exchange();
+
+    // these values are only allowed in response and not to be set via request body
+    verify(shippingInstructionService).createShippingInstruction(argument.capture());
+    assertNull(argument.getValue().getDocumentStatus());
+
+    checkStatus201.andThen(checkShippingInstructionResponseTOJsonSchema).apply(exchange);
```

Comment:
Why does this check succeed? carrierBookingReference is both set on SI (line 121) and on CargoItem (line 82) - you are only allowed to set it one of the places

Reply:
Because the actual `createShippingInstruction` method is stubbed. We are only testing the controller endpoint, but it reminds me that we need a test case for that in the ServiceImpl.

Reply:
Updated with a new check and a test case in `ShippingInstructionServiceImplTest`.